src/app/core.ts

```diff
@@ -83,6 +83,14 @@ export interface AppOptions {
    * specifying an HTTP Agent in the corresponding factory methods.
    */
   httpAgent?: Agent;
+
+  /**
+   * Prefer REST transport for Firestore.
+   * By default, Firestore will use gRPC. Pass preferRest: true to use
+   * HTTP/1.1 REST transport if possible. Note that gRPC will still be
+   * used for calls that need bi-directional streaming.
+   */
+  preferRest?: boolean;
```

Comment:
I am not entirely sure about making this an app option... is the idea to reuse the same config for other gRPC libraries in the future?

Reply:
This was just my first thought to pass it the same way as other options - alternatively, we can make it a second parameter to the `Firestore` client, or enable this via an environment variable. What would you suggest? I don't know this codebase :) I'll follow up internally to figure out the API proposal.

Reply:
I have been using the admin SDK for a couple of years and have recently changed over to using import which uses `getFirestore(appInstance)` which would only work if either an ENV variable was set, or http was set in the app instance (or add more arguments to getFirestore)

Reply:
We are actually thinking of adding the second parameter to `getFirestore`, looks like the cleanest solution. I'll get back here after we discuss internally.

Reply:
I would have expected that you'd set it the same way you'd configure the `ignoreUndefinedProperties` Firestore setting (given how it's implemented in the `@google-cloud/firestore` library)

So:
```ts
getFirestore().settings({
  preferRest: true
})
```

But if more libraries will eventually support REST over only GRPC, then I'd expect to be in the `AppOptions`, like:
```ts
initializeApp({
  preferRest: true,
})
```